[WIP] Model test spike: is this the right direction? [Y/N]

scmountain opened the pull request:
@case-eee @neight-allen, We are running into issues with our attempt to model test. Working on our Food model, we wanted to ensure our methods were interacting with the knex server the way we imagined them. Is this a wild goose chase, or are we on the right path?

scmountain committed:
attempt at model test
We are trying to test our methods on our food model, to make sure we are getting the correct
response from our server when we post/delete etc... we're using the test
to zone in on what the knex is returning when it makes a database
insertion or query.

case-eee commented:
@scmountain - it looks like you're on the right track to me! make sure you're setting up a separate test server and test database 👍

neight-allen commented on lib/controllers/foods-controller.js:
```diff
@@ -12,6 +13,22 @@ function show (request, response) {
+  var name = request.body.name
+  var calories = request.body.calories
+
+  var whatever = Food.insertFood(name, calories)
```
Make sure you chain a `.then()` on your `insertFood()`. Since it's a promise, it won't actually execute unless you `.then` it. Typically, you'd want your response to be in the then, kind of like the commented out code below.

maxglassie committed:
fixes merge conflicts

maxglassie committed:
test error fixing

The database now needs a CASCADE command with the truncate in order
to clear the entire database, rather than only the foods table.

maxglassie committed:
working on model tests

maxglassie committed:
testing a few things
